### PR TITLE
Function to download mp3 file locally

### PR DESCRIPTION
**Pull request by yoninachmany**
r @sandlerben @anoshaminai 
-Given a url, gets the file name, creates a local file with the name, and copies the data from the url to the new file

### CODE REVIEW

**Comment by sandlerben** [comment]
I'll give this a look after Anosha

File: transcription/transcription.go [hunk 1]
```diff
@@ -29,3 +32,33 @@ func msgHeaders(from string, to []string, subject string) string {
...
+
+	// TODO: check file existence first with io.IsExist
+	// Create the file
+	output, err := os.Create(fileName)
+	if err != nil {
+		return err
+	}
+	defer output.Close()
+
+	// Get the data
```
Comment (sandlerben) [inline transcription/transcription.go]: http://www.seas.upenn.edu/~cis121/current/java_style_guide.html#avoid-useless-comments

;)

File: transcription/transcription.go [hunk 2]
```diff
@@ -29,3 +32,33 @@ func msgHeaders(from string, to []string, subject string) string {
...
 	return strings.Join(msgHeaders, "\r\n")
 }
+
+// DownloadFileFromURL downloads an mp3 file locally from a url.
+func DownloadFileFromURL(url string) error {
+	// https://github.com/thbar/golang-playground/blob/master/download-files.go
+	tokens := strings.Split(url, "/")
+	fileName := tokens[len(tokens)-1]
+
+	// TODO: check file existence first with io.IsExist
```
Comment (anoshaminai) [inline transcription/transcription.go]: Yoni and I talked about this and we both agree that we don't need to do this - in what scenario would you even have two files that have the same name? What do you think @sandlerben - if so let's remove this comment

File: transcription/transcription.go [hunk 3]
```diff
@@ -29,3 +32,33 @@ func msgHeaders(from string, to []string, subject string) string {
 	msgHeaders := []string{fromHeader, toHeader, subjectHeader}
 	return strings.Join(msgHeaders, "\r\n")
 }
+
+// DownloadFileFromURL downloads an mp3 file locally from a url.
```
Comment (anoshaminai) [inline transcription/transcription.go]: DownloadFileFromURL downloads an mp3 file locally from a url of an audio file.

File: transcription/transcription.go [hunk 4]
```diff
@@ -29,3 +32,33 @@ func msgHeaders(from string, to []string, subject string) string {
...
+func DownloadFileFromURL(url string) error {
+	// https://github.com/thbar/golang-playground/blob/master/download-files.go
```
Comment (anoshaminai) [inline transcription/transcription.go]: Taken from:

File: transcription/transcription.go [hunk 5]
```diff
@@ -29,3 +32,33 @@ func msgHeaders(from string, to []string, subject string) string {
...
+	tokens := strings.Split(url, "/")
```
Comment (anoshaminai) [inline transcription/transcription.go:39]: Add comment: Extract file name info from url to name local file

File: transcription/transcription.go (hunk 1 above)
Comment (anoshaminai) [inline transcription/transcription.go]: Why is this a useless comment (esp considering the other comments) - in this piece of code we are actually extracting the file data whereas the other step are creating an output, copying over, etc - ? I'm not sure what the distinction is between useful & useless in this case. Maybe a rephrasing? "Get file contents" as opposed to get the data?

**Comment by anoshaminai** [comment]
Mostly just little questions about commenting practices, etc, but overall looks good to me. We need to find a way to test this - how?

**Comment by sandlerben** [comment]
@anoshaminai we can use mocking, but honestly this function is so slim it doesn't really require it. Since it would be so easy to write a test, I say we might as well!

**Pull request closed without merging**